napari_builtins/_tests/test_io.py

```diff
@@ -312,28 +314,37 @@ def test_add_zarr(write_spec):
     assert out[0].shape == ZARR1.shape  # type: ignore
 
 
-def test_add_zarr_1d_array_is_ignored():
+@pytest.mark.skipif(
+    parse_version(version('zarr')) > parse_version('3a0'),
+    reason='zarr 3+ does not support setting 1d data',
```

Comment:
Wait, sorry I missed your comment @Czaki, but I can't imagine this is actually true? @d-v-b got any insights as to why this fails now?

Reply:
Zarr 2 silently drop 1d data (which is tested by this test). 

Zarr 3 raises `NotImplementedError`

Reply:
i can't really tell what that test is doing -- is there a zarr bug, or is this "ignore 1D data" a napari thing?

Reply:
No, this test checks that *we* don't *read* 1D data. But zarr (v2) is perfectly able to read and write 1D arrays.

```
In [1]: import zarr

In [2]: from tempfile import TemporaryDirectory

In [3]: td = TemporaryDirectory(suffix='.zarr')

In [4]: zarr.__version__
Out[4]: '2.18.3'

In [5]: td.name
Out[5]: '/var/folders/z7/mv1jznhx59q5vc3bp09gj69m0000gn/T/tmpcq58h89x.zarr'

In [6]: z = zarr.open(td.name, mode='w')

In [7]: arr = np.random.random(10)

In [8]: z['1d'] = arr

In [9]: arr2 = np.asarray(zarr.open(td.name, mode='r')['1d'])

In [10]: np.all(arr2 == arr)
Out[10]: np.True_
```

Reply:
Looks like the issue is that creating an array with `__setitem__` on a **Group** is what doesn't work. I can make the 1D array with create_array, which is a bit clunky, but works:

```python
>>> from tempfile import TemporaryDirectory
>>> td = TemporaryDirectory(suffix='.zarr')
>>> import zarr
>>> zarr.__version__
'3.0.0b0'
>>> z = zarr.open(store=td.name, mode='w')
>>> import numpy as np
>>> arr = np.random.random(10)
>>> z['1d'] = arr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jni/micromamba/envs/z3/lib/python3.12/site-packages/zarr/core/group.py", line 1364, in __setitem__
    raise NotImplementedError
NotImplementedError
>>> type(z)
<class 'zarr.core.group.Group'>

>>> z.create_array(name='1d', shape=arr.shape, dtype=arr.dtype, data=arr)
<Array file:///var/folders/z7/mv1jznhx59q5vc3bp09gj69m0000gn/T/tmpcecbcl0a.zarr/1d shape=(10,) dtype=float64>
```

Reply:
`create_array` allows you to specify the zarr array you are creating, whereas `z['id'] = my_numpy_array` does not, which makes the latter more magical, which is why we removed the "group is a dict" API in v3. is this a problem for napari?

Reply:
haha looks like I will need to bring this up in https://github.com/zarr-developers/zarr-python/pull/2393/files

Reply:
napari won't open 1D anyways? If I try:
`viewer.add_image(np.ones(10))`
I get:
> ValueError: Image data must have at least 2 dimensions.

Reply:
We skip 1D here:
https://github.com/Czaki/napari/blob/a451f643065ce5ce0242bf491cd76a2dbc90d1c9/napari_builtins/io/_read.py#L232-L234

So then None is returned

Reply:
This test was added in #2743 to fix #1471. We should not ignore the test in v3, we should write out the 1D array and make sure again that it doesn't crash the viewer but is instead ignored.

I'll take a stab at a fix here using different code for different versions.

Reply:
fixed tests